src/runtime/parallel/coordinator.ts

```diff
@@ -47,12 +60,14 @@ export default class Coordinator {
   private nextPickleIdIndex: number
```

Comment:
Given these changes, I'm not sure the use of pickleIds and nextPickleIdIndex really makes sense anymore. Thoughts on switching making pickleIds to be editable and we splice out items as they are assigned out? We can make a copy of pickleIds and name it something like pickleIdsToExecuted if we need to retain the original list.

Reply:
I updated the code refactoring out `nextPickleIdIndex`. Looks a bit cleaner to me. I made `pickleIds` a copy of the constructor option to be safe. I don't see a need to host both the original and a copy in the coordinator at this time.